lib/util.js

```diff
@@ -622,16 +622,26 @@ function formatObject(ctx, value, recurseTimes, visibleKeys, keys) {
 
 function formatArray(ctx, value, recurseTimes, visibleKeys, keys) {
   var output = [];
-  const maxLength = Math.min(Math.max(0, ctx.maxArrayLength), value.length);
-  const remaining = value.length - maxLength;
-  for (var i = 0; i < maxLength; ++i) {
-    if (hasOwnProperty(value, String(i))) {
-      output.push(formatProperty(ctx, value, recurseTimes, visibleKeys,
-                                 String(i), true));
+  let visibleLength = 0;
+  let index = 0;
+  while (index < value.length && visibleLength < ctx.maxArrayLength) {
+    let emptyElements = 0;
+    while (index < value.length && !hasOwnProperty(value, String(index))) {
+      emptyElements++;
+      index++;
+    }
+    if (emptyElements > 0) {
+      const ending = emptyElements > 1 ? 's' : '';
+      const message = `[${emptyElements} uninitialized element${ending}]`;
+      output.push(ctx.stylize(message, 'undefined'));
     } else {
-      output.push('');
+      output.push(formatProperty(ctx, value, recurseTimes, visibleKeys,
+                                 String(index), true));
+      index++;
     }
+    visibleLength++;
   }
+  const remaining = value.length - index;
```

Comment:
No, it shouldn't. `visibleLength` is the count of... well, printed elements, i.e., `3` in case of `[ 1, [10000 uninitialized elements], 2 ]` (maybe the variable name isn't that great, though). `remaining`, on the other hand, is the count of real array indices which values were not printed.